crates/solvers/src/lib.rs

```diff
@@ -0,0 +1,12 @@
+// TODO remove this once the crate stabilizes a bit.
+#![allow(dead_code)]
+
+mod api;
+pub mod boundary;
+mod cli;
+mod config;
+mod domain;
+pub mod run;
```

Comment:
Maybe we should only reexport the `run` function from the `run` module. So we don't have to specify `crate::run::run()`. (seems a bit weird)